Static Method annotated with @DataTableType requires instance of class containing it in order to be invoked, contrary to Reflection requirements.

**Describe the bug**
When registering a DataTableType as a static method within the class which represents that Type, then when transforming a datatable to that type the given class is attempted to be instantiated by dependency injection (which is unneeded) which fails.

From the following scenarios:
```gherkin
  Scenario: Create TestBeans from a datatable transform
    Given generate TestBeans from
      | ItemOne | ItemTwo |
      | first   | second  |
      | alpha   | beta    |

  Scenario: Create TestBean directly
    Given generate TestBean from "foo" and "bar"
```

The second scenario yields this expected output:
```
Scenario: Create TestBean directly             # src/feature/TestBeans.feature:10
generated single TestBean: TestBean{stringOne='foo', stringTwo='bar'}
  Given generate TestBean from "foo" and "bar" # xxxxx.testlibrary.TestStepLibrary.generateTestBeans(java.lang.String,java.lang.String)
```

The first scenario fails:
```
Scenario: Create TestBeans from a datatable transform # src/feature/TestBeans.feature:4
  Given generate TestBeans from                       # xxxxx.testlibrary.TestStepLibrary.generateTestBeans(io.cucumber.datatable.DataTable)
      io.cucumber.datatable.CucumberDataTableException: 'java.util.List<xxxxx.testlibrary.TestBean>' could not transform
      | ItemOne | ItemTwo |
      | first   | second  |
      | alpha   | beta    |

	at io.cucumber.datatable.DataTableType.transform(DataTableType.java:144)
	at io.cucumber.datatable.DataTableTypeRegistryTableConverter.toListOrProblems(DataTableTypeRegistryTableConverter.java:151)
	at io.cucumber.datatable.DataTableTypeRegistryTableConverter.toList(DataTableTypeRegistryTableConverter.java:131)
	at io.cucumber.datatable.DataTable.asList(DataTable.java:181)
	at xxxxx.testlibrary.TestStepLibrary.generateTestBeans(TestStepLibrary.java:43)
	at ✽.generate TestBeans from(file:///E:/DevWorkspace/CucumberDataTableTransform/src/feature/TestBeans.feature:5)
Caused by: org.picocontainer.PicoCompositionException: 3 satisfiable constructors is too many for 'class java.lang.String'. Constructor List:[<init>(<init>Buffer), <init>(<init>Builder), <init>()]
	at org.picocontainer.injectors.ConstructorInjector.getGreediestSatisfiableConstructor(ConstructorInjector.java:189)
```

In the above scenarios, the TestBean class is defined as follows:
```java
package xxxxx.testlibrary;

import io.cucumber.java.DataTableType;

import java.util.Map;

public class TestBean {
    private String stringOne;
    private String stringTwo;

    public TestBean(String stringOne, String stringTwo) {
        this.stringOne = stringOne;
        this.stringTwo = stringTwo;
    }

    public String getStringOne() {
        return stringOne;
    }

    public String getStringTwo() {
        return stringTwo;
    }

    @Override
    public String toString() {
        return "TestBean{" +
                "stringOne='" + stringOne + '\'' +
                ", stringTwo='" + stringTwo + '\'' +
                '}';
    }

    @DataTableType
    public static TestBean makeTestBean(Map<String, String> row) {
        return new TestBean(
                row.get("ItemOne"),
                row.get("ItemTwo")
        );
    }
}
```


**To Reproduce**
Implement the two steps defined in the example above:
```java
package xxxxx.testlibrary;

import io.cucumber.datatable.DataTable;
import io.cucumber.java.After;
import io.cucumber.java.Before;
import io.cucumber.java.en.Given;
import io.cucumber.java.Scenario;

import java.util.List;

public class TestStepLibrary {
    protected Scenario scenario;

    @Before
    public void doBefore(Scenario scenario) {
        this.scenario = scenario;
    }

    @After
    public void doAfter(Scenario scenario) {

    }

    @Given("^generate TestBeans from$")
    public void generateTestBeans(DataTable dataTable) {
        List<TestBean> testBeans = dataTable.asList(TestBean.class);
        int count = 0;
        for(TestBean testBean : testBeans) {
            System.out.format("generated TestBean %s: %s%n", count++, testBean);
        }
    }

    @Given("^generate TestBean from \"([^\"]+)\" and \"([^\"]+)\"$")
    public void generateTestBeans(String item1, String item2) {
        TestBean testBean = new TestBean(item1, item2);
        System.out.format("generated single TestBean: %s%n", testBean);
    }
}
```


**Expected behavior**
The dataTable transform to TestBean object behaves as though the @DataTableType annotation and method were provided in a separate class, in this case the transform functions without needing to instantiate the TestBean class since the transform method is defined as a public static method.

**Context & Motivation**
We have multiple Cucumber step libraries which are pulled into various tests depending on testers' needs; most of these libraries are optional. Additionally, we also have many developers working on each library, and the simplest (and thus far easiest) guidance we've been able to provide with using dataTable transforms has been to keep both the transformed type and the method which defines that transformed type within the same class, and to ensure that the method is defined as a ```public static``` since Reflection does not require the containing class to be instantiated when calling a static method.

To facilitate that recommendation (for Cucumber 4), we have a custom method annotation which we scan for prior to launching tests (where the methods themselves are public statics, and have zero parameters), each of the methods found are added to the typeRegistry.
As a simple example, this bean class:
```java
public class DepositFromVar {
    private String toName;
    private String fromValue;

    public DepositFromVar(String toName, String fromValue) {
        this.toName = toName;
        this.fromValue = fromValue;
    }

    public String getToName() {
        return toName;
    }

    public String getFromValue() {
        return fromValue;
    }

    @CustomDataTableTypeDef
    public static DataTableType DefineDataTableType() {
        return new DataTableType(
                DepositFromVar.class,
                (Map<String, String> row) -> new DepositFromVar(
                        row.get("toName"),
                        row.get("fromValue")
                )
        );
    }
}
```

With the above bean class, the ```DefineDataTableType()``` method would be discovered by our scanner and then added to the typeRegistry (in very simplified form here):
```java
for (Method DefineDataTableTypeMethod : dataTableTypeDefMethods) {
    // Method exists, has the correct return type and parameter count
    try {
        DataTableType newDataTableType = (DataTableType) DefineDataTableTypeMethod.invoke(null);
        typeRegistry.defineDataTableType(newDataTableType);
    }
}
```

With the changes introduced in Cucumber 5, our custom scanning component is basically no longer required, however the expectation that the static method can be invoked without instantiating the class in which it is contained becomes problematic in this particular case.

This bug is not a deal-breaker or absolute blocker however since the @DataTableType annotation and attached method can still exist in a separate class, it just requires more thought and communications between our developers than they're currently used to.

**Your Environment**
- Versions used:
  - jackson-databind 2.9.8
  - cucumber libraries 4.4.0 and 5.6.0
  - junit 4.12

--- a/java/src/main/java/io/cucumber/java/AbstractGlueDefinition.java
+++ b/java/src/main/java/io/cucumber/java/AbstractGlueDefinition.java
@@ -4,13 +4,14 @@ import io.cucumber.core.backend.Located;
 import io.cucumber.core.backend.Lookup;
 
 import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
 
 import static java.util.Objects.requireNonNull;
 
 abstract class AbstractGlueDefinition implements Located {
 
     protected final Method method;
-    protected final Lookup lookup;
+    private final Lookup lookup;
     private String fullFormat;
 
     AbstractGlueDefinition(Method method, Lookup lookup) {
@@ -34,4 +35,12 @@ abstract class AbstractGlueDefinition implements Located {
         }
         return fullFormat;
     }
+
+    final Object invokeMethod(Object... args) {
+        if (Modifier.isStatic(method.getModifiers())) {
+            return Invoker.invokeStatic(this, method, args);
+        }
+        return Invoker.invoke(this, lookup.getInstance(method.getDeclaringClass()), method, args);
+    }
+
 }

--- a/java/src/main/java/io/cucumber/java/Invoker.java
+++ b/java/src/main/java/io/cucumber/java/Invoker.java
@@ -19,8 +19,16 @@ final class Invoker {
         return invoke(null, annotation, expressionMethod);
     }
 
+    static Object invokeStatic(Located located, Method method, Object... args) {
+        return doInvoke(located, null, method, args);
+    }
+
     static Object invoke(Located located, Object target, Method method, Object... args) {
         Method targetMethod = targetMethod(target, method);
+        return doInvoke(located, target, targetMethod, args);
+    }
+
+    private static Object doInvoke(Located located, Object target, Method targetMethod, Object[] args) {
         boolean accessible = targetMethod.isAccessible();
         try {
             targetMethod.setAccessible(true);

--- a/java/src/main/java/io/cucumber/java/JavaDataTableTypeDefinition.java
+++ b/java/src/main/java/io/cucumber/java/JavaDataTableTypeDefinition.java
@@ -71,28 +71,36 @@ class JavaDataTableTypeDefinition extends AbstractDatatableElementTransformerDef
         if (DataTable.class.equals(parameterType)) {
             return new DataTableType(
                 returnType,
-                (DataTable table) -> execute(replaceEmptyPatternsWithEmptyString(table))
+                (DataTable table) -> invokeMethod(
+                    replaceEmptyPatternsWithEmptyString(table)
+                )
             );
         }
 
         if (List.class.equals(parameterType)) {
             return new DataTableType(
                 returnType,
-                (List<String> row) -> execute(replaceEmptyPatternsWithEmptyString(row))
+                (List<String> row) -> invokeMethod(
+                    replaceEmptyPatternsWithEmptyString(row)
+                )
             );
         }
 
         if (Map.class.equals(parameterType)) {
             return new DataTableType(
                 returnType,
-                (Map<String, String> entry) -> execute(replaceEmptyPatternsWithEmptyString(entry))
+                (Map<String, String> entry) -> invokeMethod(
+                    replaceEmptyPatternsWithEmptyString(entry)
+                )
             );
         }
 
         if (String.class.equals(parameterType)) {
             return new DataTableType(
                 returnType,
-                (String cell) -> execute(replaceEmptyPatternsWithEmptyString(cell))
+                (String cell) -> invokeMethod(
+                    replaceEmptyPatternsWithEmptyString(cell)
+                )
             );
         }
 
@@ -102,10 +110,6 @@ class JavaDataTableTypeDefinition extends AbstractDatatableElementTransformerDef
     @Override
     public DataTableType dataTableType() {
         return dataTableType;
-    }
-
-    private Object execute(Object arg) {
-        return Invoker.invoke(this, lookup.getInstance(method.getDeclaringClass()), method, arg);
     }
 
 }

--- a/java/src/main/java/io/cucumber/java/JavaDefaultDataTableCellTransformerDefinition.java
+++ b/java/src/main/java/io/cucumber/java/JavaDefaultDataTableCellTransformerDefinition.java
@@ -16,7 +16,7 @@ class JavaDefaultDataTableCellTransformerDefinition extends AbstractDatatableEle
     JavaDefaultDataTableCellTransformerDefinition(Method method, Lookup lookup, String[] emptyPatterns) {
         super(requireValidMethod(method), lookup, emptyPatterns);
         this.transformer = (cellValue, toValueType) ->
-            execute(replaceEmptyPatternsWithEmptyString(cellValue), toValueType);
+            invokeMethod(replaceEmptyPatternsWithEmptyString(cellValue), toValueType);
     }
 
     private static Method requireValidMethod(Method method) {
@@ -52,10 +52,6 @@ class JavaDefaultDataTableCellTransformerDefinition extends AbstractDatatableEle
     @Override
     public TableCellByTypeTransformer tableCellByTypeTransformer() {
         return transformer;
-    }
-
-    private Object execute(String fromValue, Type toValueType) {
-        return Invoker.invoke(this, lookup.getInstance(method.getDeclaringClass()), method, fromValue, toValueType);
     }
 
 }

--- a/java/src/main/java/io/cucumber/java/JavaDefaultDataTableEntryTransformerDefinition.java
+++ b/java/src/main/java/io/cucumber/java/JavaDefaultDataTableEntryTransformerDefinition.java
@@ -98,7 +98,7 @@ class JavaDefaultDataTableEntryTransformerDefinition extends AbstractDatatableEl
         } else {
             args = new Object[]{fromValue, toValueType};
         }
-        return Invoker.invoke(this, lookup.getInstance(method.getDeclaringClass()), method, args);
+        return invokeMethod(args);
     }
 
 }

--- a/java/src/main/java/io/cucumber/java/JavaDocStringTypeDefinition.java
+++ b/java/src/main/java/io/cucumber/java/JavaDocStringTypeDefinition.java
@@ -18,7 +18,7 @@ class JavaDocStringTypeDefinition extends AbstractGlueDefinition implements DocS
         this.docStringType = new DocStringType(
             this.method.getReturnType(),
             contentType.isEmpty() ? method.getName() : contentType,
-            this::execute
+            this::invokeMethod
         );
     }
 
@@ -48,11 +48,6 @@ class JavaDocStringTypeDefinition extends AbstractGlueDefinition implements DocS
             .addSignature("public JsonNode json(String content)")
             .addNote("Note: JsonNode is an example of the class you want to convert content to")
             .build();
-    }
-
-
-    private Object execute(String content) {
-        return Invoker.invoke(this, lookup.getInstance(method.getDeclaringClass()), method, content);
     }
 
     @Override

--- a/java/src/main/java/io/cucumber/java/JavaHookDefinition.java
+++ b/java/src/main/java/io/cucumber/java/JavaHookDefinition.java
@@ -13,13 +13,11 @@ final class JavaHookDefinition extends AbstractGlueDefinition implements HookDef
 
     private final String tagExpression;
     private final int order;
-    private final Lookup lookup;
 
     JavaHookDefinition(Method method, String tagExpression, int order, Lookup lookup) {
         super(requireValidMethod(method), lookup);
         this.tagExpression = requireNonNull(tagExpression, "tag-expression may not be null");
         this.order = order;
-        this.lookup = lookup;
     }
 
     private static Method requireValidMethod(Method method) {
@@ -58,7 +56,7 @@ final class JavaHookDefinition extends AbstractGlueDefinition implements HookDef
             args = new Object[0];
         }
 
-        Invoker.invoke(this, lookup.getInstance(method.getDeclaringClass()), method, args);
+        invokeMethod(args);
     }
 
     @Override

--- a/java/src/main/java/io/cucumber/java/JavaParameterTypeDefinition.java
+++ b/java/src/main/java/io/cucumber/java/JavaParameterTypeDefinition.java
@@ -78,7 +78,7 @@ class JavaParameterTypeDefinition extends AbstractGlueDefinition implements Para
             args = captureGroups;
         }
 
-        return Invoker.invoke(this, lookup.getInstance(method.getDeclaringClass()), method, args);
+        return invokeMethod(args);
     }
 
 }

--- a/java/src/main/java/io/cucumber/java/JavaStepDefinition.java
+++ b/java/src/main/java/io/cucumber/java/JavaStepDefinition.java
@@ -10,8 +10,8 @@ import java.util.List;
 import static java.util.Objects.requireNonNull;
 
 final class JavaStepDefinition extends AbstractGlueDefinition implements StepDefinition {
-    private final String expression;
 
+    private final String expression;
     private final List<ParameterInfo> parameterInfos;
 
     JavaStepDefinition(Method method,
@@ -24,7 +24,7 @@ final class JavaStepDefinition extends AbstractGlueDefinition implements StepDef
 
     @Override
     public void execute(Object[] args) {
-        Invoker.invoke(this, lookup.getInstance(method.getDeclaringClass()), method, args);
+        invokeMethod(args);
     }
 
     @Override

--- a/java/src/test/java/io/cucumber/java/JavaDataTableTypeDefinitionTest.java
+++ b/java/src/test/java/io/cucumber/java/JavaDataTableTypeDefinitionTest.java
@@ -25,6 +25,13 @@ class JavaDataTableTypeDefinitionTest {
         }
     };
 
+    private final Lookup lookupForStaticMethod = new Lookup() {
+        @Override
+        public <T> T getInstance(Class<T> glueClass) {
+            throw new IllegalArgumentException("should not be invoked");
+        }
+    };
+
     private final DataTable dataTable = DataTable.create(asList(
         asList("a", "b"),
         asList("c", "d")
@@ -171,5 +178,18 @@ class JavaDataTableTypeDefinitionTest {
     public String converts_map_of_objects_to_string(Map<Object, Object> entry) {
         return "converts_map_of_objects_to_string=" + entry;
     }
+
+    @Test
+    void static_methods_are_invoked_without_a_body() throws NoSuchMethodException {
+        Method method = JavaDataTableTypeDefinitionTest.class.getMethod("static_convert_data_table_to_string", DataTable.class);
+        JavaDataTableTypeDefinition definition = new JavaDataTableTypeDefinition(method, lookupForStaticMethod, new String[0]);
+        assertThat(definition.dataTableType().transform(dataTable.asLists()), is("static_convert_data_table_to_string=[[a, b], [c, d]]"));
+    }
+
+    public static String static_convert_data_table_to_string(DataTable table) {
+        return "static_convert_data_table_to_string=" + table.cells();
+    }
+
+
 
 }

